src/sys/mman.rs

```diff
@@ -1,241 +1,251 @@
-            MAP_JIT,
-        }
-    }
-
-    pub type MmapAdvise = c_int;
-
-    pub const MADV_NORMAL     : MmapAdvise      = 0; /* No further special treatment.  */
-    pub const MADV_RANDOM     : MmapAdvise      = 1; /* Expect random page references.  */
-    pub const MADV_SEQUENTIAL : MmapAdvise      = 2; /* Expect sequential page references.  */
-    pub const MADV_WILLNEED   : MmapAdvise      = 3; /* Will need these pages.  */
-    pub const MADV_DONTNEED   : MmapAdvise      = 4; /* Don't need these pages.  */
-    pub const MADV_FREE       : MmapAdvise      = 5; /* pages unneeded, discard contents */
-    pub const MADV_ZERO_WIRED_PAGES: MmapAdvise = 6; /* zero the wired pages that have not been unwired before the entry is deleted */
-    pub const MADV_FREE_REUSABLE : MmapAdvise   = 7; /* pages can be reused (by anyone) */
-    pub const MADV_FREE_REUSE : MmapAdvise      = 8; /* caller wants to reuse those pages */
-    pub const MADV_CAN_REUSE : MmapAdvise       = 9;
-
-    libc_bitflags!{
-        pub flags MsFlags: c_int {
-            MS_ASYNC, /* [MF|SIO] return immediately */
-            MS_INVALIDATE, /* [MF|SIO] invalidate all cached data */
-            MS_KILLPAGES, /* invalidate pages, leave mapped */
-            MS_DEACTIVATE, /* deactivate pages, leave mapped */
-            MS_SYNC, /* [MF|SIO] msync synchronously */
-        }
+libc_bitflags!{
+    /// Additional parameters for `mmap()`.
+    pub flags MapFlags: c_int {
+        /// Compatibility flag. Ignored.
+        MAP_FILE,
+        /// Share this mapping. Mutually exclusive with `MAP_PRIVATE`.
+        MAP_SHARED,
+        /// Create a private copy-on-write mapping. Mutually exclusive with `MAP_SHARED`.
+        MAP_PRIVATE,
+        /// Place the mapping at exactly the address specified in `addr`.
+        MAP_FIXED,
+        /// Synonym for `MAP_ANONYMOUS`. Deprecated.
+        MAP_ANON,
+        /// The mapping is not backed by any file.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        MAP_ANONYMOUS,
+        /// Put the mapping the first 2GB of the process address space.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+        MAP_32BIT,
+        /// Used for stacks; indicates to the kernel that the mapping should extend downward in memory.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        MAP_GROWSDOWN,
+        /// Compatibility flag. Ignored.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        MAP_DENYWRITE,
+        /// Compatibility flag. Ignored.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        MAP_EXECUTABLE,
+        /// Mark the mmaped region to be locked in the same way as `mlock(2)`.
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        MAP_LOCKED,
+        /// Do not reserve swap space for this mapping.
+        MAP_NORESERVE,
```

Comment:
`MAP_NORESERVE` and `MAP_RENAME` were dropped in FreeBSD 11.  For compatibility, they're still supported in older binaries.  But there are no older binaries using these symbols from Nix.  So we should drop the symbols too, at least on FreeBSD.